net/Socket.hpp

```diff
@@ -136,13 +141,22 @@ class Socket
     static constexpr int MaximumSendBufferSize = 128 * 1024;
     static std::atomic<bool> InhibitThreadChecks;
 
-    enum Type { IPv4, IPv6, All, Unix };
+    enum class Type : uint8_t { IPv4, IPv6, All, Unix };
+    static const char* toString(Type t) noexcept;
 
     // NB. see other Socket::Socket by init below.
-    Socket(Type type)
-        : _fd(createSocket(type))
+    Socket(Type type,
+           std::chrono::steady_clock::time_point creationTime = std::chrono::steady_clock::now())
```

Comment:
I simply wasn't sure .. and in our discussion (with @mmeeks I guess) a mix of cut-off criteria were mentioned. 
The max-duration criteria, which required knowing the socket age, comes from the old `keep alive` feature (Http 1.0) and I thought it might be helpful here.

In case we deem this not required or perhaps unwarranted, then we surely can drop criteria and this creationTime time-point field. Which removes a lot of this code .. same as with max-connections ofc.

Reply:
last use case is for stats and minimum throughput criteria (currently disabled by default)